spec/spec_helper.rb

```diff
@@ -5,13 +5,15 @@
 
 RSpec.configure do |config|
 
-  config.before(:all) do
-    SparkPostRails.configure do |c|
-      c.api_key = "TESTKEY1234"
+  config.before(:each) do |example|
+    if example.metadata[:skip_configure]
+      SparkPostRails.configuration = nil # Reset configuration
+    else
+      SparkPostRails.configure do |c|
+        c.api_key = "TESTKEY1234"
+      end
     end
```

Comment:
The reason I changed this is because to test this issue in particular, I needed to ensure that a configuration object was never created (even if that means setting it to nil). If you look at the new test I created, I added a skip_configuration metadata tag since this test is a special case. I suppose I could have just set it in the test itself.